src/arch/x86/cpuid.hh

```diff
@@ -38,28 +41,41 @@ class ThreadContext;
 
 namespace X86ISA
 {
-    struct CpuidResult
-    {
-        uint64_t rax;
-        uint64_t rbx;
-        uint64_t rcx;
-        uint64_t rdx;
 
-        // These are not in alphebetical order on purpose. The order reflects
-        // how the CPUID orders the registers when it returns results.
-        CpuidResult(uint64_t _rax, uint64_t _rbx,
-                    uint64_t _rdx, uint64_t _rcx) :
-            rax(_rax), rbx(_rbx), rcx(_rcx), rdx(_rdx)
-        {}
+struct CpuidResult
+{
+    uint64_t rax;
+    uint64_t rbx;
+    uint64_t rcx;
+    uint64_t rdx;
+
+    // These are not in alphebetical order on purpose. The order reflects
+    // how the CPUID orders the registers when it returns results.
+    CpuidResult(uint64_t _rax, uint64_t _rbx,
+                uint64_t _rdx, uint64_t _rcx) :
+        rax(_rax), rbx(_rbx), rcx(_rcx), rdx(_rdx)
+    {}
 
-        CpuidResult()
-        {}
-    };
+    CpuidResult()
+    {}
+};
 
-    uint64_t stringToRegister(const char *str);
+class X86CPUID
+{
+  public:
+    X86CPUID(const X86ISAParams &p);
```

Comment:
I don't like the idea of passing the SimObject params to a non-SimObject. I know that it will make things more verbose, but I think it would be better to explicitly pass in the parameters that you need.

I'm worried about the future if/when anything about params changes. We're coupling `X86CPUID` to `BaseParams` in a way that is unnecessary.

This isn't a big problem, though.

Reply:
I originally tried to make it a SimObject but I could not figure out a way (if there is one) to create a default SimObject with all the default values. Therefore, every single python script which creates an X86 CPU would have to be changed, I think, which I wanted to avoid this way. I could potentially make the X86CPUID class in the X86ISA namespace/class.

Reply:
The class not being a SimObject isn't a problem. The (small) issue I'm raising is that I don't like that a class that is not a SimObject is taking a SimObjectParams* as the parameter to its constructor.

I would  prefer something like
```
X86CPUID::X86CPUID(std::string vendor_string, std::string name_string)
    : vendorString(vendor_string), nameString(name_string)
{...}
```
and
```
cpuid = new X86CPUID(p.vendor_string, p.name_string);
```

This way `X86CPUID` is not coupled to `X86ISAParams` but only to `X86::ISA`

Reply:
But the class is only used in X86ISA.  Doesn't your approach have the reserve problem where X86ISA depends on the enum value in X86CPUID?

Reply:
This isn't a big deal, so you can feel free to ignore.

That's exactly right. Since X86CPUID is used only in X86ISA, it's OK for X86ISA to be coupled to X86CPUID. What we don't want to couple a class that's only used in one place (X86CPUID) to a class that is outside of the same "module" (i.e., X86ISAParams).

Again, not a big deal, but I think it would be better practice to not pass around the params object. Instead of passing around the params object, I would rather see passing the explicit variable/params from that object as parameters to the class.